### PR TITLE
Fix/549 release table updates

### LINKED ISSUE #549: Update to Releases Map
## Objective
Make updates to the releases section on state pages as requested by Liz and the UCLA team. 

## Initial Request from Liz
Feedback on the table/map:
- Is it possible to figure out a way to display the full data in the third column (release details)? Either by making the table slightly wider, or wrapping the text and making the rows higher? 

- is it possible to make it more obvious that you can click on a row for more info? like something that pops up when you hover?
- could more of the additional data (the data the comes up when you click on a row) be included in the tool tip? 

## Solution Spec
1. Remove hover/highlight functionality from other tables on the site which are not clickable
2. Instead of displaying county in the first column, display the facility name, with county in smaller font underneath (this is the case with the current homepage facilities table, screenshot below). With this solution, each table entry will span two lines, providing more room to display release details. 
3. Add a tooltip that is activated when hovering over a row. The tooltip will simply say _Click row to view full details._
4. Add language in the paragraph introducing the release section that lets the user know that rows are clickable (this will be especially helpful for mobile users who likely will not see a tooltip)

### PR DESCRIPTION
**Pull request by hoshmn**
> Remove hover/highlight functionality from other tables on the site which are not clickable

@Lane / @ekmason the one table I can find that this seems to apply to is the facility table, where when a row is hovered it gets a highlighted background and its state name gets underlined to indicate the user can click the state name to view that state page. Should I remove the row highlighting there, but leave the state underlining?

I made the table very slightly wider (1rem) and reallocated more of the width to the release details column. With this done, it seems like we could just show the whole release details value rather than trying to cut it off with an ellipsis after 2 lines (which can't be done simply with CSS). Because the cell content text can never be more than a known length (it is generated by aggregating checked options in their source spreadsheet) we know the rows will never grow much -- see the following screenshot, where the Franklin row has the longest possible content (spanning 4 rows):

Screen Shot 2021-09-27 at 6 54 17 PM

If we really don't want the rows to shift slightly when a user clicks through pages of the table and there happen to be release details cells that span more than 2 rows, one option would be to set the height of all rows to a height that allows the 4 rows to show. The following shows the rows set at about 18px taller than they currently are (which cuts slightly into the padding of rows that have the very longest release details, but we could make them slightly taller if we don't want to cut into that padding):

Screen Shot 2021-09-27 at 6 14 22 PM

Here's what the tooltips look like:

Screen Shot 2021-09-27 at 7 09 04 PM

Lastly, certain rows don't have any facility name mentioned. I figured I'd put in a default to show in those cases, maybe "Multiple facilities" (not sure if that's what is implied by the facility missing?).

Thoughts?

fixes #549

### CODE REVIEW

File: src/components/states/sections/ReleasesTable.js [hunk 1]
```diff
@@ -46,10 +52,26 @@ const ReleasesTable = ({ classes, data, lang, ...props }) => {
       {
         Header: lang.table.jurisdiction,
         accessor: "jurisdiction",
-        Cell: (prop) => prop.value,
+        Cell: (prop) => {
+          const { jurisdiction, facility } = prop.row.original;
+          return (
+            <>
+              <Typography className={classes.facilityName} variant="body1">
+                {facility || "Multiple facilities"}
```
Comment (hoshmn) [inline src/components/states/sections/ReleasesTable.js:60]: todo: add to CMS if we do go with a default here

**Comment by ekmason** [comment]
> Should I remove the row highlighting there, but leave the state underlining?

Since this functionality was pretty recently implemented (this was part of Phase 2, right?) and Liz approved, I'd lean toward keeping it - I'll defer to @Lane on this of course! 

> Because the cell content text can never be more than a known length (it is generated by aggregating checked options in their source spreadsheet) we know the rows will never grow much

Is this because the field is comprised of members of a closed option set? I'll check with Liz on this, but couldn't that option set grow if UCLA thought it necessary? 

>Lastly, certain rows don't have any facility name mentioned. I figured I'd put in a default to show in those cases, maybe "Multiple facilities" (not sure if that's what is implied by the facility missing?).

I'll raise this to Liz today to get a sense of what behavior she'd like to see here. 

--

I'm noticing that as the table gets longer, the legend shifts down with it - thus distancing it from the map. Would it be possible to keep the legend just beneath the map despite the length of the table?

**Comment by Lane** [comment]
thanks @hoshmn and @ekmason!  a few additional notes.

>  Should I remove the row highlighting there, but leave the state underlining?

yes, the row isn't clickable so it shouldn't have a hover state.  the state link is the clickable item, so it should have a hover state when hovered, but it should not have a hover state when the row is hovered.

> show full release details in table

i'm not a fan of the variable height rows because they cause jumping between pages, but given it's not a huge jump it's fine with me.

> facility names

can we adjust styling of these to match other table rows where we have a primary + secondary value:

image

**Comment by hoshmn** [comment]
thanks for the feedback both @Lane & @ekmason ! i made those adjustments. Eliana good catch on the legend getting further than ideal from the state -- unfortunately i don't this is a product of the table getting taller but rather happens when a state has a higher width:height ratio and so it doesn't extend to the bottom of its allotted svg container, leaving a gap between it and the legend. i don't think there's any easy fix for that :\

**Comment by Lane** [comment]
@hoshmn can you set `position: sticky` on the map wrapper to keep the map in view within the section?  for some states, when you get to the table part of the state can be scrolled out of view:

current behaviour:

image


desired behaviour:

**Comment by Lane** [comment]
@hoshmn thanks, sticky looks good!

noticing the tooltip is coming up above the hovered row instead of to the right on the preview:

image

also noticing that there is no hovered state on the table rows.  we want to keep the hover state on rows that are clickable (have actions tied to them) but remove from tables where clicking on the row doesn't do anything.

**Comment by hoshmn** [comment]
@Lane 
> noticing the tooltip is coming up above the hovered row instead of to the right on the preview:

moved it back - moved it to "top" because that's where they go for hovered facility names for the HomeTable
Screen Shot 2021-10-04 at 4 52 15 PM
and, if set to "right" and there isn't room to the right, they obscure the beginning of the row
Screen Shot 2021-10-04 at 4 53 50 PM
but I'm fine with whatever you think is best


> also noticing that there is no hovered state on the table rows. we want to keep the hover state on rows that are clickable (have actions tied to them) but remove from tables where clicking on the row doesn't do anything.

whoops! added the hover state back for tables passed an `onRowClick` handler

**Comment by Lane** [comment]
gotcha, we want to keep these "click row" tooltips out of the data area, so right works best i think.

not too concerned about the placement on the smaller resolutions because those will mainly be touch devices without tooltips. 

thanks for those changes, merging in!

**Pull request merged**